symengine/sets.cpp

```diff
@@ -419,6 +471,19 @@ RCP<const Set> FiniteSet::set_union(const RCP<const Set> &o) const
             }
         }
     }
+    if (is_a<Reals>(*o)) {
+        set_basic container;
+        for (const auto &elem : container_) {
+            if (is_a<Complex>(*elem)) {
```

Comment:
Instead of checking if the value is a complex number here, we should check if the value is not a real number. For eg: symbols should be considered complex since we don't know what the value is going to be.